### PR TITLE
Adding missing data to TreasurySummary

### PR DESCRIPTION
**Pull request by franjs**
Adding missing data, refactoring typeDefs and formatting balances

### CODE REVIEW

File: subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts [hunk 1]
```diff
@@ -28,19 +30,41 @@ export async function treasurySummary(
...
+      frozenMisc: formatBalance(api, treasuryBalance.frozenMisc),
+      reservedBalance: formatBalance(api, treasuryBalance.reservedBalance),
+      votingBalance: formatBalance(api, treasuryBalance.votingBalance),
     },
-    burn: burn?.toString(),
+    nextBurn: formatBalance(api, burn ?? BN_ZERO, true),
+  };
+}
+
+function createSpendPeriod(api: Context['api'], bestNumber: BlockNumber): SpendPeriod {
```
Comment (ishrak-imam) [inline subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts:49]: Should this go in the `substrateChainService`?

File: subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts [hunk 2]
```diff
@@ -28,19 +30,41 @@ export async function treasurySummary(
...
+    spendPeriod: createSpendPeriod(api, bestNumber),
     treasuryBalance: {
       accountId: treasuryBalance.accountId.toString(),
       accountNonce: treasuryBalance.accountNonce.toString(),
-      freeBalance: treasuryBalance.freeBalance.toString(),
-      frozenFee: treasuryBalance.frozenFee.toString(),
-      frozenMisc: treasuryBalance.frozenMisc.toString(),
-      reservedBalance: treasuryBalance.reservedBalance.toString(),
-      votingBalance: treasuryBalance.votingBalance.toString(),
+      freeBalance: formatBalance(api, treasuryBalance.freeBalance),
```
Comment (jdomingos) [inline subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts:39]: I did something similar with the format balance but instead passing the registry. 
So doing `formatBalance` X amount of times, aren't you query the API those amount of times? 

Wouldn't it be better to get the registry **once** and pass the registry to the function? 
Like so:
https://github.com/litentry/litentry-graph/pull/109/files#diff-efaf21b3700988dcc51fc2f45e46b102278079fede4a0922dfcd718ce1cf366cR77

Do you have any thoughts on that? :)

**Review by jdomingos — COMMENTED** [review]
Looks good overall but actually have a question about the `formatBalance` 😄

File: subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts (hunk 2 above)
Comment (jdomingos) [reply subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts:39]: Maybe `api.registry` is not an async call and I've overengineered 🤔

File: subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts (hunk 1 above)
Comment (franjs) [reply subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts:49]: I don't think we will use this format in other places, the value of the spendPeriod itself may be used in the bounties but not in this shape. We can move it later if needed

File: subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts (hunk 2 above)
Comment (franjs) [reply subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts:39]: `api.registry` does not make any network call, it's a constant created when the api is initialized

**Review by ishrak-imam — APPROVED** [review]

**Review by jdomingos — APPROVED** [review]
LGTM! thanks for answering my question

**Pull request merged**